### PR TITLE
Wait until scaled methods, do not fail if the resource is gone.

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicaSetOperationsImpl.java [hunk 1]
```diff
@@ -126,15 +126,25 @@ public ReplicaSet scale(int count, boolean wait) {
...
         try {
-          ReplicaSet rc = getMandatory();
+          ReplicaSet rc = get();
+          //If the rs is gone, we shouldn't wait.
+          if (rc == null) {
+            if (count == 0) {
+              countDownLatch.countDown();
+              return;
+            } else {
+              return;
```
Comment (jstrachan) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicaSetOperationsImpl.java]: so we still block forever if we try to scale up and the resource has gone now right? Why not pass an exception back to the caller

Comment (iocanel) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicaSetOperationsImpl.java]: I agree in the long run we should just pass the exception back to the caller.

Comment (jstrachan) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicaSetOperationsImpl.java]: this is separate to passing the exception - by returning and not counting down the latch we'll block the caller thread forever right (exception or no exception)?

**Comment by jstrachan** [comment]
[merge]

**Pull request merged**